### PR TITLE
Removes 8.x RNs and What's new from main

### LINKED ISSUE #6041: Remove `8.x` release notes from `main`
`main` = `9.0` 

Related to: https://github.com/elastic/docs-projects/issues/257

Creating this issue as a reminder to clear release notes, breaking changes, and what's new highlights from the `8.x` series. 

We need to do this before the `9.0.0-beta1` scheduled for feature freeze on Jan 15, 2025.

### PR DESCRIPTION
**Pull request by benironside**
Fixes #6041 as part of #[257](https://github.com/elastic/docs-projects/issues/257)

8.x release notes, as well as what's new, need to be removed from main in preparation for 9.0.

### CODE REVIEW

**Review by nastasha-solomon — APPROVED** [review]
Thanks for doing this @benironside! The local preview looked good and didn't contain any leftover What's new files/pages or 8.x release notes. Approving early so it's ready to merge once the CI checks pass.

File: docs/release-notes.asciidoc [hunk 1]
```diff
@@ -1,87 +1,5 @@
 [[release-notes]]
+[chapter]
```
Comment (nastasha-solomon) [inline docs/release-notes.asciidoc:2]: @benironside you might need to remove this if you re-add the what's new page.

File: docs/index.asciidoc [hunk 2]
```diff
@@ -8,7 +8,7 @@ include::{asciidoc-dir}/../../shared/attributes.asciidoc[]
 
 include::es-overview.asciidoc[]
 
-include::whats-new.asciidoc[]
+include::whats-new.asscidoc[]
```
Comment (benironside) [inline docs/index.asciidoc:11]:
```suggestion
include::whats-new.asciidoc[]
```

File: docs/release-notes.asciidoc [hunk 3]
```diff
@@ -2,86 +2,3 @@
 = Release notes
```
Comment (benironside) [inline docs/release-notes.asciidoc:2]:
```suggestion
[chapter]
= Release notes
```

**Comment by benironside** [comment]
run docs-build

**Review by nastasha-solomon — APPROVED** [review]
Thanks, everything looks good @benironside! It's interesting that two `chapter` tags (namely the new one added [here](https://github.com/elastic/security-docs/pull/6246/files#diff-1eabae06876833e37a07c4851b77a13fd4ddb924e7c6cea6d241b39e20723c19R2)) doesn't affect the CI checks or TOC structure at all. I don't know enough about the book tags though to suggest a change though, so if it works, I don't want to mess with it anymore. Anywho, looks good to merge!

**Pull request merged**